New on the web June (#8251)
GoogleChrome/web.dev: src/site/content/en/blog/web-platform-06-2022/index.md

{% BrowserCompat 'api.ReadableStreamBYOBReader' %}

### Access locally installed fonts

Chrome 103 includes the [Local Font Access API](/local-fonts/), which allows access to the user's locally installed fonts. After requesting access to the fonts installed on the device, call `window.queryLocalFonts()` to get an array of the installed fonts.

```js
const opts = {};
const pickedFonts = await self.queryLocalFonts();
for (const fontData of pickedFonts) {
  console.log(fontData.postscriptName);
  console.log(fontData.fullName);
  console.log(fontData.family);
  console.log(fontData.style);
}
```

### The update media feature

Firefox 102 includes the [`update`](https://developer.mozilla.org/docs/Web/CSS/@media/update-frequency) media feature. This is used to query whether the output device can modify the appearance of content once it has been rendered.

{% BrowserCompat 'css.at-rules.media.update' %}

### A new HTTP status code—103 early hints

> Chrome 103 adds a new status code HTTP 103 Early Hints. If the server knows that a certain set of subresources is required to load a page, it can preload those resources as the page that requires them comes in. This requires updates to your server to take advanage of the feature, [find out more](https://developer.chrome.com/blog/early-hints/). 

Reviewer: A few comments about this one (this may be a bit nit picky so feel free to ignore some of these):
- One of key use cases of Early Hints is for CDNs rather than origin servers so the term "server" may not get this across.
- Preconnect is another important use case as well as preload - see the Shopify example note in that blog post that made waves on Twitter.
- The server does not preload - it advises the browser to do this if it needs to - a big differentiator of this over HTTP/2 Push that this is being seen as a better replacement for.
- Noticed a typo in "advanage" (this should be fixed, even if the other points are ignored as being nit picky).

```suggestion
Chrome 103 adds a new status code HTTP 103 Early Hints. If the server or CDN knows that a certain set of subresources is required to load a page, it can advise the browser to preconnect to origins or even preload resources as the page that requires them comes in. This requires updates to your server or CDN to take advantage of the feature, [find out more](https://developer.chrome.com/blog/early-hints/). 
```

Reviewer: Actually one more point. Should we avoid generic link text like "find out more", and instead go with "find out more about Early Hints"? Updated above suggested change with that.

## Beta browser releases

Beta browser versions give you a preview of things that will be in the next stable version of the browser. It's a great time to test new features, or removals, that could impact your site before the world gets that release.

New betas in April were [Chrome 104](https://blog.chromium.org/2022/06/chrome-104-beta-new-media-query-syntax.html), [Firefox 103](https://developer.mozilla.org/docs/Mozilla/Firefox/Releases/103), and [Safari 16](https://developer.apple.com/documentation/safari-release-notes/safari-16-release-notes). 

> Chrome 103 includes the new syntax for [range media queries](https://developer.chrome.com/blog/media-query-range-syntax/), from the Media Queries Level 4 specification. For example, a media query previously written like this:

Author: ```suggestion
Chrome 104 includes the new syntax for [range media queries](https://developer.chrome.com/blog/media-query-range-syntax/), from the Media Queries Level 4 specification. For example, a media query previously written like this:
```
```suggestion
Chrome 103 includes the new syntax for [range media queries](https://developer.chrome.com/blog/media-query-range-syntax/), from the Media Queries Level 4 specification. For example, a media query previously written like this:
```

---
layout: post
title: "New to the web platform in June"
subhead: >
  Discover some of the interesting features that landed in stable and beta web browsers during June 2022. 
description: >
  Discover some of the interesting features that landed in stable and beta web browsers during June 2022.
date: 2022-06-30
hero: image/kheDArv5csY6rvQUJDbWRscckLr1/xPDclET3NIAnmeBygeTW.jpg
alt: The word new on a red brick wall.
authors:
  - rachelandrew
tags:
  - blog
  - new-to-the-web
---

## Stable browser releases

In June, [Chrome 103](https://developer.chrome.com/blog/new-in-chrome-103/) and [Firefox 102](https://developer.mozilla.org/docs/Mozilla/Firefox/Releases/102) became stable.

### Transform streams and readable byte streams

> Firefox 102 includes support for [Transform Streams](https://developer.mozilla.org/docs/Web/API/TransformStream). This enables piping from ReadableStream to a WritableStream, executing a transformation on the chunks. It's great to see this feature become available in all three engines, making this a very good time to learn about [Streams](/streams/).

Reviewer 2: ```suggestion
Firefox 102 includes support for [Transform Streams](https://developer.mozilla.org/docs/Web/API/TransformStream). This enables piping from `ReadableStream` to a `WritableStream`, executing a transformation on the chunks. It's great to see this feature become available in all three engines, making this a very good time to learn about [Streams](/streams/).
```

{% BrowserCompat 'api.TransformStream' %}

> [Readable byte streams](https://developer.mozilla.org/docs/Web/API/Streams_API#bytestream-related_interfaces) are also now supported in Firefox 102, enabling a BYOB (bring your own buffer) reader with the [ReadableStreamBYOBReader](https://developer.mozilla.org/docs/Web/API/ReadableStreamBYOBReader) interface. This can be used to stream data supplied by the developer. 

Reviewer 2: ```suggestion
[Readable byte streams](https://developer.mozilla.org/docs/Web/API/Streams_API#bytestream-related_interfaces) are also now supported in Firefox 102, enabling a BYOB (bring your own buffer) reader with the [`ReadableStreamBYOBReader`](https://developer.mozilla.org/docs/Web/API/ReadableStreamBYOBReader) interface. This can be used to stream data supplied by the developer. 
```

{% BrowserCompat 'api.ReadableStreamBYOBReader' %}

### Access locally installed fonts

Chrome 103 includes the [Local Font Access API](/local-fonts/), which allows access to the user's locally installed fonts. After requesting access to the fonts installed on the device, call `window.queryLocalFonts()` to get an array of the installed fonts.

```js
const opts = {};
const pickedFonts = await self.queryLocalFonts();
for (const fontData of pickedFonts) {
  console.log(fontData.postscriptName);
  console.log(fontData.fullName);
  console.log(fontData.family);
  console.log(fontData.style);
}
```

> ### The update media feature

Reviewer 3: ```suggestion
### The `update` media feature
```
